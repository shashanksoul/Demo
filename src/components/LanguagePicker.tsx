import React, {Dispatch, useState} from 'react';
import {StyleSheet, View} from 'react-native';
import DropDownPicker from 'react-native-dropdown-picker';
import lnList from './../localization/translations/languageList.json';
import i18next from './../localization';

const LanguagePicker = () => {
  const [open, setOpen] = useState(false);
  const [lng, setLng] = useState('en');
  const lngList = Object.keys(lnList).map((key: string) => ({
    label: lnList[key].nativeName,
    value: key,
  }));

  const setValue = (val: any) => {
    setLng(val);
    i18next.changeLanguage(val);
  };
  return (
    <>
      <DropDownPicker
        testID="picker"
        open={open}
        value={lng}
        items={lngList}
        setOpen={setOpen}
        setValue={callback => setValue(callback(lng))}
        dropDownContainerStyle={style.container}
        style={style.container}
      />
    </>
  );
};

const style = StyleSheet.create({
  container: {
    width: 100,
    borderRadius: 25,
    textAlign: 'left',
  },
});

export default LanguagePicker;
