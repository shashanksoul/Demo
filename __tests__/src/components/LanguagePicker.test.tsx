import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import LanguagePicker from '../../../src/components/LanguagePicker';
import i18next from 'i18next';

describe('Language Picker component', () => {
  it('render', () => {
    const picker = render(<LanguagePicker />);
    expect(picker).toMatchSnapshot();
  });

  it('change language', () => {
    const {getByTestId} = render(<LanguagePicker />);
    const picker = getByTestId('picker');
    fireEvent(picker, 'setValue', () => 'hi');
    expect(i18next.language).toEqual('hi');
  });
});
