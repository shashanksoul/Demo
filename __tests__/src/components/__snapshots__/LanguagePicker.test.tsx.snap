// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Language Picker component render 1`] = `
<View
  style={
    [
      {
        "width": "100%",
      },
      {
        "zIndex": 5000,
      },
      {},
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onLayout={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderColor": "#000",
        "borderRadius": 25,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "minHeight": 50,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 3,
        "textAlign": "left",
        "width": 100,
        "zIndex": 5000,
      }
    }
    testID="picker"
  >
    <Text
      style={
        [
          {
            "color": "#000",
            "flex": 1,
          },
          {},
          {},
          false,
        ]
      }
    >
      English
    </Text>
    <View
      style={
        [
          {
            "marginLeft": 10,
          },
          {},
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../react-native-dropdown-picker/src/themes/light/icons/arrow-down.png",
          }
        }
        style={
          [
            {
              "height": 20,
              "width": 20,
            },
            {},
          ]
        }
      />
    </View>
  </View>
</View>
`;
